server/handlers/project_command_output_handler.go

```diff
@@ -0,0 +1,134 @@
+package handlers
+
+import (
+	"sync"
+
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+type DefaultProjectCommandOutputHandler struct {
+	// this is TerraformOutputChan
+	ProjectCmdOutput chan *models.ProjectCmdOutputLine
+	// this logBuffers
+	projectOutputBuffers map[string][]string
+	// this is wsChans
+	controllerBuffers map[string]map[chan string]bool
+	// same as chanLock
+	controllerBufferLock sync.RWMutex
+	logger               logging.SimpleLogging
+}
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_project_command_output_handler.go ProjectCommandOutputHandler
+
+type ProjectCommandOutputHandler interface {
+	// Send will enqueue the msg and wait for Handle() to receive the message.
+	// this will be called in the RunCommandAsync where we currently sending messages to the terraform channel
+	Send(ctx models.ProjectCommandContext, msg string)
+
+	// Receive will create a channel for projectPullInfo and run a callback function argument when the new channel receives a message. This will be called from the controller.
+	// j.ProjectCommandOutputHandler.Receive(pull, func(msg string) {
+	//   j.Logger.Info(msg)
+	//   if err := c.WriteMessage(websocket.BinaryMessage, []byte(msg+"\r\n\t")); err != nil {
+	//     j.Logger.Warn("Failed to write ws message: %s", err)
+	//     return err
+	//   }
+	// })
+	Receive(projectInfo string, callback func(msg string) error) error
+
+	// Is basically the same as Listen function from logstreaming controller.
+	Handle()
+}
+
+func NewProjectCommandOutputHandler() ProjectCommandOutputHandler {
+	return &DefaultProjectCommandOutputHandler{
+		ProjectCmdOutput: make(chan *models.ProjectCmdOutputLine),
+	}
+}
```

Comment:
```suggestion
func NewProjectCommandOutputHandler(projectCmdOutput chan *models.ProjectCmdOutputLine, logger logging.SimpleLogging) ProjectCommandOutputHandler {
    return &DefaultProjectCommandOutputHandler{
        ProjectCmdOutput: projectCmdOutput,
        logger: logger,
        controllerBuffer: map[string]map[chan string]bool{},
    }
}
```
Use this function in the `server.go`